pkg/api/v1/types.go

```diff
@@ -145,3 +148,22 @@ type ServicePrincipalProfile struct {
 	ClientID string `json:"clientId,omitempty"`
 	Secret   string `json:"secret,omitempty"`
 }
+
+// AuthProfile defines all possible authentication profiles for OpenShift cluster
+type AuthProfile struct {
+	IdentityProviders []IdentityProvider `json:"identityProviders,omitempty"`
+}
+
+// IdentityProvider is heavily cut down equivalent to IdentityProvider in the upstream
+type IdentityProvider struct {
+	Name     string      `json:"name"`
+	Provider interface{} `json:"provider,omityempty"`
+}
+
+// AADIdentityProvider defines Identity provider for MS AAD.
+// it is based on OpenID IdentityProvider
+type AADIdentityProvider struct {
+	Kind     string `json:"kind,omitempty"`
```

Comment:
Since providers are typed to struct is `kind` needed? E.g. can customer set provider to `AADIdentityProvider` and `kind` to `Github` (will need more validation to prevent this from happening)

Reply:
Kind is used to Unmarshal details to right struct. As potentially "Github" auth provider will have a different structure. They can make "kind: Github", but it will fail validation, or If they provide wrong fields in the Struct itself (`Kind: AADIdentityProvider` and data for Github) it will still be validated out for them.

At the moment only one provider supported. But I think for GA we will need to add more, so we building capability already.